md/bdm-access-control.md

```diff
@@ -314,3 +315,157 @@ Deploy your access control file.
+    - **Content** : requestList (script, click the fx icon to switch from contstant to script)
+    - **Column keys** : persistenceId, subject, content, medicalComment, student.fullname
+    
+Now in the Studio, create a new [application](applicationCreation.md). Call it *TeacherApp*, give it a theme, and a homepage token, and map it to the *Teacher* profile.
+Then create a one-page menu *Student Request*, with your homepage token and add to it the page you just created in the UID.
+
+If you preview the page at this point, both jan.fischer and helen.kelly can access your new application, select from a drop down list either *Mathematics* or *Physics*, wich should display all the Mathematics and all the Physics requests, respectively, for both users.
+
+**5. Security implementation**
+
+- Got to *BonitaStudioSubscription-7.7.0/workspace/tomcat/setup/*
+- Run *setup pull*
+- Go the */BonitaStudioSubscription-7.7.0/workspace/tomcat/setup/platform_conf/current/tenants/1/tenant_portal/* folder that should have just appeared.
+- Open the *dynamic-permissions-checks-custom.properties*, and add the following line:
+``` GET|bdm/businessData/com.company.model.Request=[check|SubjectTeacherPermissionRule] ```
+- This line indicates that before executing any queries on the com.company.model.Request object types in the BDM, a verification has to be run. In this case we indicate that the verification is a groovy script, *SubjectTeacherPermissionRule.groovy*, which we will now create.
+- Go to */BonitaStudioSubscription-7.7.0/workspace/tomcat/setup/platform_conf/current/tenants/1/tenant_security_scripts*. Create a file *SubjectTeacherPermissionRule.groovy*, with the following content:
+
+```groovy
+
+import org.bonitasoft.engine.api.APIAccessor
+import org.bonitasoft.engine.api.Logger
+import org.bonitasoft.engine.api.permission.APICallContext
+import org.bonitasoft.engine.api.permission.PermissionRule
+import org.bonitasoft.engine.profile.Profile
+import org.bonitasoft.engine.profile.ProfileCriterion
+import org.bonitasoft.engine.session.APISession
+
+class SubjectTeacherPermissionRule implements PermissionRule {
+
+    @Override
+    boolean isAllowed(APISession apiSession, APICallContext apiCallContext, APIAccessor apiAccessor, Logger logger) {
+        APISession session = apiSession
+        long currentUserId = session.getUserId()
+        List<Profile> profilesForUser = apiAccessor.getProfileAPI().getProfilesForUser(currentUserId, 0, 50, ProfileCriterion.ID_ASC)
+        // First, let's check we only restrict access to query named "findBySubject":
+        if (!apiCallContext.getQueryString().contains("q=findBySubject")) {
+            return true
+        }
+
+        def filters = apiCallContext.getFilters()
+        if (filters.containsKey("subject")) {
+            def subjectAsString = filters.get("subject")
+            //Let's check the logged-in user (teacher) has at least one profile matching the Class Subject:
```

Comment:
maybe add a small example on the comment to help the user to understand the logic: 
subjectAsString == Physics 
profile PhysicsTeacher contains physics -> ok
profile MathematicsTeacher doesn't contain Physics -> not ok
=> only someone with the profile PhysicsTeacher will have th authorization